// ngx-loader-spinner/esm5/lib/ngx-loader-spinner.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
var NgxLoaderSpinnerService = /** @class */ (function () {
    function NgxLoaderSpinnerService() {
        /**
         * Loading
         */
        this.loading = new BehaviorSubject(false);
    }
    /**
     * Show loader
     */
    NgxLoaderSpinnerService.prototype.show = function () {
        this.loading.next(true);
    };
    /**
     * Hide loader
     */
    NgxLoaderSpinnerService.prototype.hide = function () {
        this.loading.next(false);
    };
    /**
     * Loader status
     */
    NgxLoaderSpinnerService.prototype.loaderStatus = function () {
        return this.loading;
    };
    NgxLoaderSpinnerService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxLoaderSpinnerService_Factory() { return new NgxLoaderSpinnerService(); }, token: NgxLoaderSpinnerService, providedIn: "root" });
    NgxLoaderSpinnerService = __decorate([
        Injectable({
            providedIn: 'root'
        })
    ], NgxLoaderSpinnerService);
    return NgxLoaderSpinnerService;
}());
export { NgxLoaderSpinnerService };
