// ngx-loader-spinner/esm5/lib/ngx-loader-spinner.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { NgxLoaderSpinnerService } from './ngx-loader-spinner.service';
var NgxLoaderSpinnerComponent = /** @class */ (function () {
    function NgxLoaderSpinnerComponent(loader) {
        this.loader = loader;
        /**
         * Loader flag
         */
        this.isLoading = false;
    }
    NgxLoaderSpinnerComponent.prototype.ngOnInit = function () {
        this.getLoaderStatus();
    };
    /**
     * Get loader status
     */
    NgxLoaderSpinnerComponent.prototype.getLoaderStatus = function () {
        var _this = this;
        this.loader
            .loaderStatus()
            .subscribe(function (spinning) {
            setTimeout(function () {
                _this.isLoading = spinning;
            }, 10);
        });
    };
    NgxLoaderSpinnerComponent.ctorParameters = function () { return [
        { type: NgxLoaderSpinnerService }
    ]; };
    NgxLoaderSpinnerComponent = __decorate([
        Component({
            selector: 'ngx-loader-spinner',
            template: "\n  <div *ngIf=\"isLoading\" class=\"loader-wrapper\">\n    <div class=\"loader\">\n      <div class=\"item1\"></div>\n      <div class=\"item2\"></div>\n      <div class=\"item3\"></div>\n    </div>\n  </div>\n  ",
            styles: ["\n    .loader-wrapper {\n      height: 100vh;\n      width: 100vw;\n      background-color: rgba(255, 255, 255, 0.7);\n      position: fixed;\n      z-index: 999;\n      top: 0;\n      left: 0;\n      right: 0;\n      bottom: 0;\n      display: flex;\n      align-items: center;\n      justify-content: center;\n    }\n    .loader > div {\n      width: 18px;\n      height: 18px;\n      background-color: #d35400;\n      border-radius: 100%;\n      display: inline-block;\n      -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;\n      animation: sk-bouncedelay 1.4s infinite ease-in-out both;\n    }\n    .loader .item1 {\n      -webkit-animation-delay: -0.32s;\n      animation-delay: -0.32s;\n    }\n    .loader .item2 {\n      -webkit-animation-delay: -0.16s;\n      animation-delay: -0.16s;\n    }\n\n    @-webkit-keyframes sk-bouncedelay {\n      0%, 80%, 100% { -webkit-transform: scale(0) }\n      40% { -webkit-transform: scale(1.0) }\n    }\n\n    @keyframes sk-bouncedelay {\n      0%, 80%, 100% {\n        -webkit-transform: scale(0);\n        transform: scale(0);\n      } 40% {\n        -webkit-transform: scale(1.0);\n        transform: scale(1.0);\n      }\n    }\n  "]
        })
    ], NgxLoaderSpinnerComponent);
    return NgxLoaderSpinnerComponent;
}());
export { NgxLoaderSpinnerComponent };
